src/idl_gen_cpp.cpp

```diff
@@ -2282,9 +2282,15 @@ class CppGenerator : public BaseGenerator {
     // Definition for type traits for this table type. This allows querying var-
     // ious compile-time traits of the table.
     if (opts_.g_cpp_std >= cpp::CPP_STD_17) {
+      auto file_identifier = parser_.file_identifier_.length()
+                                 ? ('"' + parser_.file_identifier_ + '"')
+                                 : "nullptr";
       code_ += "struct {{STRUCT_NAME}}::Traits {";
       code_ += "  using type = {{STRUCT_NAME}};";
       code_ += "  static auto constexpr Create = Create{{STRUCT_NAME}};";
+      code_ +=
+          "  static const char* constexpr FileIdentifier = " + file_identifier +
```

Comment:
Why is this `const char*` but in the generated files it is `auto`?

Reply:
Because the other flatbuffers API uses either a non-zero `const char*` or `nullptr` in the API (e.g., `Finish`). Now `auto FileIdentifier = nullptr` deduces to `std::nullptr_t`. This would result in a different type of `FileIdentifier` based on its value, which is not a good design for a traits class, in my opinion.

Reply:
Wait, I misinterpreted your question. The answer is way simpler: my fault. 🙂

Let me update the PR.

Reply:
Use `*` aligned to right.
Use `FLATBUFFERS_CONSTEXPR` since we support compilers that don't know `constexpr`.

Reply:
Ah I see its only generated for C++17.. why not also for earlier versions?

Reply:
I'll move the pointer over.

The whole code block is wrapped in a condition

```cpp
if (opts_.g_cpp_std >= cpp::CPP_STD_17)`
```

It seems that it's only generated for C++17. Should I still use `FLATBUFFERS_CONSTEXPR`? The existing `Create` function doesn't use it either.